clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -119,8 +130,14 @@ void SYCL::constructLLVMForeachCommand(Compilation &C, const JobAction &JA,
   SmallString<128> ForeachPath(C.getDriver().Dir);
   llvm::sys::path::append(ForeachPath, "llvm-foreach");
   const char *Foreach = C.getArgs().MakeArgString(ForeachPath);
-  C.addCommand(std::make_unique<Command>(JA, *T, ResponseFileSupport::None(),
-                                         Foreach, ForeachArgs, None));
+
+  auto Cmd = std::make_unique<Command>(JA, *T, ResponseFileSupport::None(),
+                                       Foreach, ForeachArgs, None);
+  // FIXME: Add the FPGA specific timing diagnostic to the foreach call.
+  // The foreach call obscures the return codes from the tool it is calling
+  // to the compiler itself.
```

Comment:
To clarify, is it the behavior of `llvm-foreach` that should FIXed?

Reply:
Right - I have already made the needed adjustments to pass back the error code from the tool being called through `llvm-foreach` but having to add the diagnostic code check to both the tool call and `llvm-foreach` is not efficient and can present issues if we have multiple tools using the same return code, and those tools go through `llvm-foreach`